woocommerce/woocommerce
# Add drag & drop offline payment settings

### Changes proposed in this Pull Request:

Slack: p1731047871781339/1731031068.064989-slack-C07B44M2B6K

Partially addresses https://github.com/woocommerce/woocommerce/issues/52053:

- Adds `Sortable` component to handle drag and drop
- Adds drag & drop capability to offline payment methods screen

Initially, I tried fixing the issues with existing `Sortable` component from `@woocommerce/components` but I wasn't able to find out the root cause of the component's drag issues. Ultimately, I ended up using a `dndkit` since it's also used by blocks package.

#### Not done in PR

- Add drag & drop capability to main payments screen due to ongoing discussion p6riRB-bfm-p2#comment-11488
- Add backend persistence due to ongoing discussion in above point. Instead, I temporarily use localStorage to establish a basic data structure
 


### How to test the changes in this Pull Request:

1. Enable `reactify-classic-payments-settings` feature flag
2. Go to `WooCommerce > Settings > Payments`
3. Click on `Offline payment gateways`
4. Drag the drag handle and attempt to change ordering
5. Observe you're able to re-order payment method
6. Refresh the screen and observe the ordering is persisted




https://github.com/user-attachments/assets/cfc91302-3ebf-44e6-9d95-fb1b80ee6b2e



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
